lib/workflow/stores/mongo.js

```diff
@@ -994,7 +995,7 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _, san
     exports.updateGraphProgress = function(data) {
         assert.uuid(data.graphId, "Graph ID should be a UUID");
         assert.string(data.progress.description, "Progress info should include description");
```

Comment:
This is not in this PR, but maybe try ```assert.object(data.progress)``` before assert the members of data.progress. If data.progress is undefined, you will get ```TypeError: Cannot read property 'description' of undefined``` exception and the assertion will not work.